src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -10,29 +10,77 @@
 using Akka.Actor;
 using Akka.Persistence;
 using Akka.Util.Internal;
+using System.Threading.Tasks;
 
 namespace Akka.Cluster.Sharding
 {
     using ShardId = String;
     using EntryId = String;
     using Msg = Object;
 
+    public class PersistentShardActor : PersistentActor, IPersistentActorContext
```

Comment:
Because now there are two different actors with the similar logic. 
There is 
- `ShardActor` - for shards without remembering entities
- `PersistentShardActor` - for shards with remembering entities using persistence
- maybe in the future also `DDataShardActor` - for shards with remembering entities using DData

The logic is now extracted into `Shard` and `PersistentShard` with inheritance and those actors only sends incoming messages into the logic object